File: w4/cat_rental_requests/app/models/cat.rb
require 'action_view'

class Cat < ActiveRecord::Base
  include ActionView::Helpers::DateHelper

  CAT_COLORS = %w(black white orange brown)

  has_many(
    :rental_requests,
    class_name: "CatRentalRequest",
    dependent: :destroy
  )

  validates(
    :birth_date,
    :color,
    :name,
    :sex,
    presence: true
  )

  belongs_to(
    :owner,
    foreign_key: :user_id,
    primary_key: :id,
    class_name: 'User'
  )

  validates :color, inclusion: CAT_COLORS
  validates :sex, inclusion: %w(M F)

  def age
    time_ago_in_words(birth_date)
  end
end

File: w5/js_exercises/cat.js
function Cat(owner, name) {
  this.name = name;
  this.owner = owner;
  this.cuteStatement = function () {
    return this.owner + " loves " + this.name;
  }
};

var gizmo = new Cat("me", "gizmo");
console.log(gizmo);
